src/lib/OpenEXRCore/openexr_priv_memory.h

```diff
@@ -0,0 +1,14 @@
+/*
+** SPDX-License-Identifier: BSD-3-Clause
+** Copyright Contributors to the OpenEXR Project.
+*/
+
+#ifndef OPENEXR_PRIVATE_MEMORY_H
+#define OPENEXR_PRIVATE_MEMORY_H
+
+#include "openexr_base.h"
+
+extern void *priv_alloc( size_t bytes ) __attribute__((malloc));
```

Comment:
The attribute is surely gcc/clang only and won't work on MSVS, right?

Reply:
oh, quite possibly, I don't have a windows machine at home, so I end up using CI to discover these :)

Reply:
MSVC doesn't have an equivalent, I think. Am I reading the gcc docs right, that the attribute says "a returned pointer has no objects embedded in it, so a realloc is safe? I'm curious about what kind of optimizations it enables gcc to do in response? It might be worth making it gcc specific. As of apple clang 12, clang has started to lose attributes that were implemented for compatibility. Ultimately Apple will upstream these, and clang will diverge from gcc in the end.

Reply:
my understanding is that it is two fold: one, it allows the optimizer know that the memory is not aliased, so it doesn't have to reload from other non-aliased stores, which realloc can not not.

However, the real reason I attached it was that I was under the impression that the various static analyzers would use this to report further down where memory was requested that the memory was not freed, so it would report the call site, not the routine